static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx

```diff
@@ -218,6 +218,42 @@ function isJavascriptSDKTransaction(transaction: TraceTree.Transaction): boolean
   );
 }
 
+function isPageloadTransaction(node: TraceTreeNode<TraceTree.NodeValue>): boolean {
+  return isTransactionNode(node) && node.value['transaction.op'] === 'pageload';
+}
+
+function isSSRTransaction(node: TraceTreeNode<TraceTree.NodeValue>): boolean {
+  return isTransactionNode(node) && node.value['transaction.op'] === 'http.server';
+}
+
+function isBrowserRequestSpan(value: TraceTree.Span): boolean {
+  return value.op === 'browser' && value.description === 'request';
+}
+
+/**
+ * Swaps the two nodes in the graph.
+ */
+function childParentSwap({
```

Comment:
This is a bit more annoying than I would like. Tldr, but because the child nodes keep a reference to the parent, it means we need to filter out the children part of the tree so that we dont end up with circular references.